rust/tool/src/esp.rs

```diff
@@ -1,10 +1,53 @@
 use std::array::IntoIter;
 use std::path::{Path, PathBuf};
 
-use anyhow::{Context, Result};
+use anyhow::{Context, Result, bail};
 
 use crate::generation::Generation;
 
+/// Supported system
+#[allow(dead_code)]
+#[non_exhaustive]
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub enum Architecture {
```

Comment:
This struct encodes a lot of information from differnte domains. We should make this more generic, move it out of the `esp.rs` module and with only one function besides the constructor: `efi_representation()` (or another appropriate name).  This function would be similar to https://github.com/systemd/systemd/blob/main/meson.build#L2114  The caller (i.e. in the esp module) then has the domain kowledge to assemble the filenames just from this information.

Reply:
This is now done, I used a Ext-style pattern for assembling the filenames based on the domain knowledge, is that fine with you?